pkg/standalone-utils/test/PermitRelayer.test.ts

```diff
@@ -0,0 +1,432 @@
+import { ethers } from 'hardhat';
+import { Contract } from 'ethers';
+import { expect } from 'chai';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+
+import Token from '@balancer-labs/v2-helpers/src/models/tokens/Token';
+import TokenList from '@balancer-labs/v2-helpers/src/models/tokens/TokenList';
+import { GeneralPool } from '@balancer-labs/v2-helpers/src/models/vault/pools';
+import { encodeExit, encodeJoin } from '@balancer-labs/v2-helpers/src/models/pools/mockPool';
+import {
+  encodeCalldataAuthorization,
+  signBatchSwapAuthorization,
+  signExitAuthorization,
+  signJoinAuthorization,
+  signPermit,
+  signSwapAuthorization,
+} from '@balancer-labs/v2-helpers/src/models/misc/signatures';
+
+import { deploy } from '@balancer-labs/v2-helpers/src/contract';
+import * as expectEvent from '@balancer-labs/v2-helpers/src/test/expectEvent';
+import { actionId } from '@balancer-labs/v2-helpers/src/models/misc/actions';
+
+import { bn } from '@balancer-labs/v2-helpers/src/numbers';
+import { MAX_UINT256, ZERO_ADDRESS } from '@balancer-labs/v2-helpers/src/constants';
+
+const tokenInitialBalance = bn(200e18);
+
+const setup = async () => {
+  const [, admin, lp] = await ethers.getSigners();
+
+  const dai = await deploy('TestTokenPermitDAI', { args: [admin.address, 'DAI', 'DAI', 18] });
+  const mkr = await deploy('TestTokenPermit', { args: [admin.address, 'MKR', 'MKR', 18] });
+
+  const tokenList = [await Token.fromContract(dai), await Token.fromContract(mkr), await Token.create('WETH')];
+  const tokens = new TokenList(tokenList.sort((a, b) => a.compare(b)));
+
+  // Deploy Balancer Vault
+  const authorizer = await deploy('v2-vault/Authorizer', { args: [admin.address] });
+  const vault = await deploy('v2-vault/Vault', { args: [authorizer.address, tokens.WETH.address, 0, 0] });
```

Comment:
If we remove `TestTokenPermit` (and just make that `TestToken`), we should be able to simplifly this greatly by using TokenList.create and Vault.create.

Reply:
Ah, I wasn't aware of Vault.create. I'll change to use that.

`TokenList.create` is a little trickier as we need to handle `TestDAI` as well and I don't really want to add both styles of permit to `TestToken`

Reply:
These also don't really play nicely together as I'll need to extract WETH from the vault and create a `Token` manually for it.

Reply:
I think you're right. Let's leave as is here then, and get back to it in the future if we encounter this again (which I doubt, as I don't think we'll need to test DAI's custom permit often).